yt_dlp/extractor/echo360.py

```diff
@@ -0,0 +1,118 @@
+import re
+
+from .common import InfoExtractor
+from ..utils import (
+    determine_ext,
+    float_or_none,
+    traverse_obj,
+    variadic,
+)
+
+
+class Echo360IE(InfoExtractor):
+    _INSTANCES_RE = r'''(?:
+                            echo360\.ca|
+                            echo360\.net\.au|
+                            echo360\.org\.au|
+                            echo360\.org\.uk|
+                            echo360\.org|
+                        )'''
+    _UUID_RE = r'[\da-fA-F]{8}-[\da-fA-F]{4}-[\da-fA-F]{4}-[\da-fA-F]{4}-[\da-fA-F]{12}'
+    _VALID_URL = rf'''(?x)
+        https?://(?P<host>{_INSTANCES_RE})
+        /media/(?P<id>{_UUID_RE})/public'''
+
+    _API_BASE = 'https://%s/api/ui/echoplayer/public-links/%s/media/%s/player-properties'
+
+    _TESTS = [
+        {
+            'url': 'https://echo360.org.uk/media/1d8392aa-a3e7-4e78-94cf-b6532c27208c/public',
+            'info_dict': {
+                'id': '3c7ae6e0-fa19-432d-aa21-c283b4276f2a',
+                'ext': 'mp4',
+                'title': '3-4 Force + moment + mechanics.mp4',
+                'duration': 4731.888,
+            },
+            'params': {'skip_download': 'm3u8'}
+        },
+        {
+            'url': 'https://echo360.net.au/media/f04960a9-2efc-4b63-87b5-72e629081d15/public',
+            'info_dict': {
+                'id': '6098a147-2d65-40f3-b9e9-a0204afe450c',
+                'ext': 'mp4',
+                'title': 'EXSC634_Online_Workshop_Week_4.mp4',
+                'duration': 6659.72,
+            },
+            'params': {'skip_download': 'm3u8'}
+        },
+    ]
+
+    def _call_api(self, host, video_id, media_id, session_token, **kwargs):
+        return self._download_json(
+            self._API_BASE % (host, video_id, media_id), video_id,
+            headers={'Authorization': f'Bearer {session_token}'}, **kwargs)
+
+    @staticmethod
+    def _update_url_query(uri, query_string):
+        if query_string is not None:
+            return f'{uri.split("?", 1)[0]}?{query_string}'
+        return uri
+
+    @staticmethod
+    def _get_query_string(uri, query_strings):
+        uri_base = uri.split("?", 1)[0]
+        for query_string in query_strings:
+            if re.match(query_string['uriPattern'], uri_base):
+                return query_string['queryString']
+        return None
+
+    def _parse_mediapackage(self, video):
+        video_id = video['playableAudioVideo']['mediaId']
+        query_strings = traverse_obj(video, ('sourceQueryStrings', 'queryStrings')) or []
+
+        formats = []
+        for track in variadic(traverse_obj(video, ('playableAudioVideo', 'playableMedias')) or []):
```

Comment:
Why variadic?
```suggestion
        for track in traverse_obj(video, ('playableAudioVideo', 'playableMedias', ...)):
```

Reply:
I copied it from my opencast extractor, I thought someone suggested that on purpose. But I also think it is not necessary. This suggestion looks clean.